File: core-java-practice/leet-code-codebase/ReverseTheNumber.java
import java.util.*;
public class reverse_the_no {
  public static void main(String[] args) {
    Scanner sc=new Scanner(System.in);
    int n=sc.nextInt();
    int rn=0;
    //loop for reversing the number with the logic
    for(int i=0;i<=n;i++){
      int ld=n%10;
      rn=rn*10+ld;
      n=n/10;
    }
    System.out.println(rn);
  }
}

File: core-java-practice/leet-code-codebase/ReverseTheNumber.cs
using System;

class ReverseTheNumber{
    static void Main(){
        Console.WriteLine("Enter a number:");
        int n = int.Parse(Console.ReadLine()); // Read input
        int rn = 0; // variable to store reversed number
        int temp = n; // store original value to avoid losing n
        // Loop for reversing the number
        while (temp > 0){
            int ld = temp % 10;   // get last digit
            rn = rn * 10 + ld;    // build reversed number
            temp = temp / 10;     // remove last digit
        }
        Console.WriteLine("Reversed number: " + rn);
    }
}
